medical-portal/src/API/RSBC.DMF.MedicalPortal.API/Controllers/DocumentController.cs

```diff
@@ -130,6 +142,84 @@ public async Task<ActionResult> GetAllDocuments([FromRoute] string driverId)
+        [HttpPost("upload")]     
+        [ProducesResponseType(typeof(OkResult), 200)]
+        [ProducesResponseType(401)]
+        [ProducesResponseType(500)]
+        [ActionName(nameof(UploadDriverDocument))]
+        // NOTE an operation filter was started but the generated code was not completed
+        // to save time, we opted to just write this api call manually on the frontend
+        [ApiExplorerSettings(IgnoreApi = true)]
+
+        public async Task<IActionResult> UploadDriverDocument([FromForm] IFormFile file, [FromForm] int documentSubTypeId, [FromForm] string driverId)
+        {
+            // sanity check paramters
+            if (string.IsNullOrEmpty(file?.FileName) || file.Length == 0)
+            {
+                return BadRequest();
+            }
+
+            // validate document sub type
+            var documentIdRequest = new DocumentIdRequest();
+            documentIdRequest.Id = documentSubTypeId;
+            var documentSubTypeGuidReply = _documentManagerClient.GetDocumentSubTypeGuid(documentIdRequest);
+            if (documentSubTypeGuidReply.ResultStatus != Rsbc.Dmf.CaseManagement.Service.ResultStatus.Success)
+            {
+                _logger.LogError("ERROR getting document sub type id.\n {0}", documentSubTypeGuidReply.ErrorDetail);
+                return StatusCode(500, documentSubTypeGuidReply.ErrorDetail);
+            }
+
+            var profile = await _userService.GetCurrentUserContext();
+
+            // read file stream into byte array
+            var fileStream = new MemoryStream((int)file.Length);
+            file.CopyTo(fileStream);
+            var bytes = fileStream.ToArray();
+
+            // verify mimetype from byte array
+            var mimeTypes = new MimeTypes();
+            var mimeType = mimeTypes.GetMimeType(bytes);
+            if (!DocumentUtils.IsAllowedMimeType(mimeType.Name))
+            {
+                _logger.LogError($"ERROR in uploading file due to invalid mime type {mimeType?.Name}");
+                return BadRequest("Invalid file type.");
+            }
+
+            // add the document
+            var request = new UploadFileRequest()
+            {
+                ContentType = file.ContentType,
+                Data = ByteString.CopyFrom(bytes),
+                EntityName = "dfp_driver",
+                FileName = file.FileName,
+                FolderName = driverId,
+            };
+            var fileReply = _documentStorageAdapterClient.UploadFile(request);
+            if (fileReply.ResultStatus != Pssg.DocumentStorageAdapter.ResultStatus.Success)
+            {
+                return StatusCode(500, fileReply.ErrorDetail);
+            }
+
+            // create document and then link to driver
+            var driver = new Driver();
+            driver.Id = driverId;
+
+            var document = _documentFactory.Create(driver, profile.Id, fileReply.FileName, "Submitted Document", _configuration["DRIVER_DOCUMENT_TYPE_CODE"]);
```

Comment:
optional: this is my bad but we should replace "Submitted Document" with a constant or enum